pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -641,148 +642,116 @@ func (s *PluginSuite) TestBundlePrune() {
-
 	// Tainted public key on raw format
-	certPublicKeyRaw, err := x509.MarshalPKIXPublicKey(s.cert.PublicKey)
-	require.NoError(t, err)
-
-	// Create new bundle with two certs
-	bundle := bundleutil.BundleProtoFromRootCAs("spiffe://foo", []*x509.Certificate{s.cert, s.cacert})
-	bundle.X509TaintedKeys = []*common.X509TaintedKey{
-		{PublicKey: []byte("foh")},
-	}
+	skID := x509util.SubjectKeyIDToString(s.cert.SubjectKeyId)
 
-	// Bundle not found
-	err = s.ds.TaintX509CA(ctx, "spiffe://foo", unusedKey.Public())
-	spiretest.RequireGRPCStatus(t, err, codes.NotFound, _notFoundErrMsg)
+	t.Run("bundle not found", func(t *testing.T) {
+		err := s.ds.TaintX509CA(ctx, "spiffe://foo", "foo")
+		spiretest.RequireGRPCStatus(t, err, codes.NotFound, _notFoundErrMsg)
+	})
 
-	_, err = s.ds.CreateBundle(ctx, bundle)
+	// Create Malformed CA
+	bundle := bundleutil.BundleProtoFromRootCAs("spiffe://foo", []*x509.Certificate{{Raw: []byte("bar")}})
+	_, err := s.ds.CreateBundle(ctx, bundle)
 	require.NoError(t, err)
 
-	// Bundle contains a malformed tainted key
-	err = s.ds.TaintX509CA(ctx, "spiffe://foo", unusedKey.Public())
-	spiretest.RequireGRPCStatusHasPrefix(t, err, codes.Internal, "failed to parse tainted Key:")
+	t.Run("bundle not found", func(t *testing.T) {
+		err := s.ds.TaintX509CA(ctx, "spiffe://foo", "foo")
+		spiretest.RequireGRPCStatus(t, err, codes.Internal, "failed to parse rootCA: x509: malformed certificate")
+	})
 
-	// Remove malformed tainted key
-	bundle.X509TaintedKeys = []*common.X509TaintedKey{}
+	// Update bundle
+	bundle = bundleutil.BundleProtoFromRootCAs("spiffe://foo", []*x509.Certificate{s.cert, s.cacert})
 	_, err = s.ds.UpdateBundle(ctx, bundle, nil)
 	require.NoError(t, err)
 
-	// Invalid public key to taint provided
-	err = s.ds.TaintX509CA(ctx, "spiffe://foo", unusedKey)
-	spiretest.RequireGRPCStatus(t, err, codes.InvalidArgument, "failed to marshal public key to taint: x509: unsupported public key type: *ecdsa.PrivateKey")
-
-	// Taint successfully
-	err = s.ds.TaintX509CA(ctx, "spiffe://foo", s.cert.PublicKey)
-	require.NoError(t, err)
+	t.Run("taint successfully", func(t *testing.T) {
+		err := s.ds.TaintX509CA(ctx, "spiffe://foo", skID)
+		require.NoError(t, err)
 
-	fetchedBundle, err := s.ds.FetchBundle(ctx, "spiffe://foo")
-	require.NoError(t, err)
+		fetchedBundle, err := s.ds.FetchBundle(ctx, "spiffe://foo")
+		require.NoError(t, err)
 
-	expectedRootCAs := []*common.Certificate{
-		{DerBytes: s.cert.Raw},
-		{DerBytes: s.cacert.Raw},
-	}
+		expectedRootCAs := []*common.Certificate{
+			{DerBytes: s.cert.Raw, TaintedKey: true},
+			{DerBytes: s.cacert.Raw},
+		}
 
-	require.Equal(t, expectedRootCAs, fetchedBundle.RootCas)
+		require.Equal(t, expectedRootCAs, fetchedBundle.RootCas)
+	})
 
-	expectedTaintedKeys := []*common.X509TaintedKey{
-		{PublicKey: certPublicKeyRaw},
-	}
-	require.Equal(t, expectedTaintedKeys, fetchedBundle.X509TaintedKeys)
+	t.Run("no bundle with provided skID", func(t *testing.T) {
```

Comment:
It would be good to check that the sequence number of the bundle remains the same for those cases that fail.